paprika/setup.py

```diff
@@ -31,732 +26,685 @@ def _get_installed_benchmarks():
+        )
+
+        # Remove the COM from the host coordinates to make alignment easier.
+        structure.coordinates -= center_of_mass
+        host_structure.coordinates -= center_of_mass
+
+        # Find the principal components of the host, take the two largest, and find
+        # the vector orthogonal to that. Use that vector to align with the z-axis.
+        # This may not generalize to non-radially-symmetric host molecules.
+        inertia_tensor = np.dot(
+            host_structure.coordinates.transpose(), host_structure.coordinates
+        )
+
+        eigenvalues, eigenvectors = np.linalg.eig(inertia_tensor)
+        order = np.argsort(eigenvalues)
+
+        _, axis_2, axis_1 = eigenvectors[:, order].transpose()
+
+        cavity_axis = np.cross(axis_1, axis_2)
+
+        # Add dummy atoms which will be used to align the structure.
+        cls.add_dummy_atoms_to_structure(structure, [np.array([0, 0, 0]), cavity_axis])
+
+        # Give atoms uniform mass so that the align code uses the center
+        # of geometry rather than the center of mass.
+        for atom in structure.atoms:
+            atom.mass = 1.0
+
+        aligned_structure = align.zalign(structure, ":DM1", ":DM2")
+
+        # Return a copy of the original structure where things like the masses
+        # have not been changed and dummy atoms not added.
+
+        # noinspection PyTypeChecker
+        structure: pmd.Structure = pmd.load_file(coordinate_path, structure=True)
+        structure.coordinates = aligned_structure["!:DM1&!:DM2"].coordinates
+
+        return structure
+
+    @classmethod
+    def prepare_complex_structure(
+        cls,
+        coordinate_path: str,
+        guest_atom_indices: List[int],
+        guest_orientation_mask: str,
+        pull_distance: float,
+        pull_window_index: int,
+        n_pull_windows: int,
+    ) -> pmd.Structure:
+        """Prepares the coordinates of both the host and the guest molecule in a
+        given APR window.
+
+        Parameters
+        ----------
+        coordinate_path
+            The path to the coordinate file which contains the guest molecule
+            bound to the host.
+        guest_atom_indices
+        guest_orientation_mask
+        pull_distance
+        pull_window_index
+        n_pull_windows
         """
-        Read the YAML recipe for the host and guest.
+
+        # Align the host-guest complex so the first guest atom is at (0, 0, 0) and the
+        # second guest atom lies along the positive z-axis.
+        # noinspection PyTypeChecker
+        structure: pmd.Structure = pmd.load_file(coordinate_path, structure=True)
+
+        (
+            guest_orientation_mask_0,
+            guest_orientation_mask_1,
+        ) = guest_orientation_mask.split(" ")
+
+        aligned_structure = align.zalign(
+            structure, guest_orientation_mask_0, guest_orientation_mask_1
+        )
```

Comment:
@jeff231li can you check that these new alignment routines (here and around line 82) will still work with your workflows? It's hard for me to tell at a glance if this a rewriting of the same procedure or if this procedure is a little different than what we had before.

Reply:
It's a slightly different procedure from what I have, the method here uses two dummy atoms placed along the cavity axis as input to `zalign`. Whereas what I have applies a rotation matrix to the structure based on the angle between the largest principal axes vector and a user input vector (default is z-axis:(0,0,1)). Both methods can achieve the same thing here.

Reply:
> It's hard for me to tell at a glance if this a rewriting of the same procedure or if this procedure is a little different than what we had before.

This is just a refactor of the code taken from the master branch, so should produce the same results as that.

Reply:
@jeff231li it seems the changes you are talking about are still in another branch, so probably best to merge these changes first, then update your feature branch with these changes.

Reply:
Yep, agree.

Reply:
Also agree.